org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/core/compiler/IProblem.java

```diff
@@ -829,25 +829,35 @@ public interface IProblem {
 	int AbstractMethodsInConcreteClass = TypeRelated + 333;
 
 	/** @deprecated - problem is no longer generated, use {@link #UndefinedType} instead */
+	@Deprecated
```

Comment:
when those constants are not used anymore, can you please add "forRemoval"?
like
`@Deprecated(forRemoval = true, since = "3.37.0 2024-03")`

Reply:
The deprecation (in javadoc) is actually much older (older than the annotation type? likely older than "forRemoval" :) ) - should we still use the current version in since?

Reply:
> the since refers to the forRemoval

No, 
> [String](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/String.html) since
Returns the version in which the annotated element became deprecated.

And
> boolean forRemoval
Indicates whether the annotated element is subject to removal in a future version.

There is no direct connection between both properties.

Reply:
@jukzi once more I scanned the constants in question, and found that those have been deprecated some time in the range 2005 - 2011. So proper marking would require to look up the exact JDT release for each of those constants to provide a proper `since` value. Setting `forRemoval` is a separate strategic decision altogether.

In this PR I only want to make the existing deprecations (javadoc) visible during a test (using the annotation). Can't we think of this as orthogonal to the issue of improving the details of deprecation? This PR should serve only one purpose: avoid ambiguous IDs of "active" problems, ignoring those that are no longer raised.

Reply:
My desire is to actually have unused code to be removed in the future. That should only be done 2 years after announcing the removal. For that purpose it does not matter when exactly deprecation happened, but it should have the forRemoval flag 2 years in front. If you do not feel confident about adding a rough date it is fine if you just ignore my request to add a date (or the forRemoval at all) and leave that burden for someone else in the future.  In doubt we just live with the unused constants forever. Wont be of any problem. 
Still the easiest and widely accepted solution (yes, slightly dirty, but most helpful) would be to temporarily just use the current's release version/date and delete it after 2 years.